### PR TITLE
Schedule modal short-term accessibility improvements

### PR DESCRIPTION
**Pull request by thecristen**
Asana ticket: https://app.asana.com/0/385363666817452/1200074880846056/f
~Hoping that a few improved labels will help things! We're testing on dev blue with our user.~ Testing was successful! Opening this PR up for code review here.

### CODE REVIEW

File: apps/site/assets/ts/schedule/components/schedule-finder/daily-schedule/DailySchedule.tsx [hunk 1]
```diff
@@ -72,39 +72,41 @@ const SchedulesSelect = ({
 
   return (
     <div className="schedule-finder__service-selector">
-      <label htmlFor="service_selector" className="sr-only">
-        Schedules
+      <label>
```
Comment (thecristen) [inline apps/site/assets/ts/schedule/components/schedule-finder/daily-schedule/DailySchedule.tsx:75]: Moving the `<select>` to _inside_ the associated `<label>` seems to work more reliably than using the `htmlFor` attribute, so I decided to try that here.

File: apps/site/assets/ts/schedule/components/schedule-finder/daily-schedule/DailySchedule.tsx [hunk 2]
```diff
@@ -173,21 +175,29 @@ export const DailySchedule = ({
...
-              origin: stopId,
-              direction: directionId,
-              date: selectedService!.end_date
-            }}
-          />
+      <div id="daily-schedule">
+        {isLoading(fetchState) ? (
+          <Loading />
+        ) : (
+          <span className="sr-only" aria-live="polite">
```
Comment (thecristen) [inline apps/site/assets/ts/schedule/components/schedule-finder/daily-schedule/DailySchedule.tsx:182]: `aria-live` announces dynamic changes to screen readers!

File: apps/site/assets/ts/schedule/components/schedule-finder/daily-schedule/DailySchedule.tsx [hunk 3]
```diff
@@ -72,39 +72,41 @@ const SchedulesSelect = ({
...
+          Choose a schedule type from the available options
+        </span>
+        <SelectContainer>
+          <select
+            className="c-select-custom text-center u-bold"
+            defaultValue={defaultSelectedServiceId}
+            onChange={e =>
+              onSelectService(sortedServices.find(s => s.id === e.target.value))
+            }
+            aria-controls="daily-schedule"
```
Comment (thecristen) [inline apps/site/assets/ts/schedule/components/schedule-finder/daily-schedule/DailySchedule.tsx:86]: This attribute explicitly indicates that this `<select>` controls the daily schedules section.

**Review by alicia-mbta — APPROVED** [review]
`<span aria-love="lots">I love all these improvements` ❤️ `</span>`

**Review by hannahpurcell — APPROVED** [review]
Yeah, I think these changes are excellent!  Glad Ian approved the workflow, because I'm still learning to use a screen reader 😅

**Pull request merged**